RecoTracker/MeasurementDet/plugins/TkStackMeasurementDet.cc

```diff
@@ -24,12 +33,103 @@ TkStackMeasurementDet::RecHitContainer TkStackMeasurementDet::recHits(const Traj
+      return result;
+    }
+    unsigned int indexl = cil - begin;
+    LogTrace("MeasurementTracker") << " index cluster lower" << indexl << " on detId "
+                                   << fastGeomDet().geographicalId().rawId();
+
+    for (const_iterator ciu = upperDetSet.begin(); ciu != upperDetSet.end(); ++ciu) {
+      unsigned int indexu = ciu - begin;
+      if (ciu < begin) {
+        edm::LogError("IndexMisMatch") << "TkStackMeasurementDet cannot create hit because of index mismatch.";
+        return result;
+      }
+      LogTrace("VectorHitBuilderAlgorithm") << " index cluster upper " << indexu;
+
+      if (data.phase2OTClustersToSkip().empty() or
+          ((not data.phase2OTClustersToSkip()[indexl]) and (not data.phase2OTClustersToSkip()[indexu]))) {
+        Phase2TrackerCluster1DRef clusterLower = edmNew::makeRefTo(data.phase2OTData().handle(), cil);
+        Phase2TrackerCluster1DRef clusterUpper = edmNew::makeRefTo(data.phase2OTData().handle(), ciu);
+        //ERICA:I would have prefer to keep buildVectorHits ...
+        VectorHit vh = theMatcher->buildVectorHit(&specificGeomDet(), clusterLower, clusterUpper);
+        LogTrace("MeasurementTracker") << "TkStackMeasurementDet::rechits adding VectorHits!" << std::endl;
+        LogTrace("MeasurementTracker") << vh << std::endl;
+        result.push_back(std::make_shared<VectorHit>(vh));
+      }
+    }
+  }
+  
+
+*/
+
+  //unsigned int rawDetId1(lowerDet()->index());
+  //DetId detId1(rawDetId1);
+  //DetId detIdStack = theTkTopo->stack(detId1);
+  //DetId detIdStack = theTkTopo->stack(specificGeomDet().geographicalId());
+  DetId detIdStack = specificGeomDet().geographicalId();
+
+  //std::cout << "ID of current stack " << detIdStack << std::endl;
+  auto iterator = data.phase2OTVectorHits().find(detIdStack);
+  if (iterator == data.phase2OTVectorHits().end())
+    return result;
+  for (auto& vecHit : data.phase2OTVectorHits()[detIdStack])
+    result.push_back(std::make_shared<VectorHit>(vecHit));
   return result;
 }
 
 bool TkStackMeasurementDet::measurements(const TrajectoryStateOnSurface& stateOnThisDet,
                                          const MeasurementEstimator& est,
                                          const MeasurementTrackerEvent& data,
                                          TempMeasurements& result) const {
-  return true;
+  LogDebug("MeasurementTracker") << "TkStackMeasurementDet::measurements";
+
+  if (!isActive(data)) {
+    result.add(theInactiveHit, 0.F);
+    return true;
+  }
+
+  LogTrace("MeasurementTracker") << " is active";
+
+  auto oldSize = result.size();
+  MeasurementDet::RecHitContainer&& allHits = recHits(stateOnThisDet, data);
+
+  for (auto&& hit : allHits) {
+    std::pair<bool, double> diffEst = est.estimate(stateOnThisDet, *hit);
+    LogDebug("MeasurementTracker") << "New vh added with chi2: " << diffEst.second;
```

Comment:
shouldn't this be inside the if statement below?

Reply:
Makes sense, yes. I moved it.